pennylane/transforms/optimization/merge_amplitude_embedding.py

```diff
@@ -22,7 +25,235 @@
+                    outvals = custom_handler(self, *invals, **eqn.params)
+                elif getattr(primitive, "prim_type", "") == "operator":
+                    outvals = self.interpret_operation_eqn(eqn)
+                elif getattr(primitive, "prim_type", "") == "measurement":
+                    self.merge_and_purge_new_operations()
+                    outvals = self.interpret_measurement_eqn(eqn)
+                else:
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    extra_args, params = primitive.get_bind_params(eqn.params)
+                    outvals = primitive.bind(*extra_args, *invals, **params)
+
+                if not primitive.multiple_results:
+                    outvals = [outvals]
+                for outvar, outval in zip(eqn.outvars, outvals, strict=True):
+                    self._env[outvar] = outval
+
+            # The following is needed because any operations inside self.new_operations have not yet
+            # been applied. At this point, we **know** that any operations that should be merged
+            # have been merged, and operations left inside self.new_operations should be applied
+            self.merge_and_purge_new_operations()
+
+            # Read the final result of the Jaxpr from the environment
+            outvals = []
+            for var in jaxpr.outvars:
+                outval = self.read(var)
+                if isinstance(outval, qml.operation.Operator):
+                    outvals.append(super().interpret_operation(outval))
+                else:
+                    outvals.append(outval)
+
+            self.cleanup()
+            self._env = {}
+            return outvals
+
+    # Overwrite the cond primitive so that wire collision can be handled
+    # correctly across the different branches
+    @MergeAmplitudeEmbeddingInterpreter.register_primitive(cond_prim)
+    def _(self, *invals, jaxpr_branches, consts_slices, args_slice):
+        args = invals[args_slice]
+
+        new_jaxprs = []
+        new_consts = []
+        new_consts_slices = []
+        end_const_ind = len(jaxpr_branches)
+
+        # Store seen wires before we begin to process the branches
+        original_wires = self.state.get("visited_wires", set())
+
+        for const_slice, jaxpr in zip(consts_slices, jaxpr_branches):
+            consts = invals[const_slice]
+            if jaxpr is None:
+                new_jaxprs.append(None)
+                new_consts_slices.append(slice(0, 0))
+            else:
+                new_jaxpr = jaxpr_to_jaxpr(copy(self), jaxpr, consts, *args)
+
+                # Store newly visited wires
+                newly_seen_wires = self.state.get("visited_wires", set()) - original_wires
+                # Reset visited wires for next branch so we don't get false positive collisions
+                # between cond branches
+                self.state["visited_wires"] = set()
```

Comment:
This shouldn't be an empty set. It should be `original_wires`. You will probably also have to make `original_wires` a copy of `visited_wires` rather than the same set.